### DISCUSSION TITLE
Google MAP Overlay (Google Earth)

### DISCUSSION THREAD

**Discussion by neoalius** [Mr-MIBonk/M.I.B._More-Incredible-Bash#209] [Ideas]
Can you please look on the possibility of making google earth work? 

Vehicle details: 
A4 2017
MMI Firmware: MHI2_ER_AUG22_K3346
VC firmware: AU_C1_AU491_0379

![IMG_0062](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/14fda9d1-a4c1-4aee-a489-c33fa716451f)
![IMG_0063](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/0f211e27-efa7-48bc-a071-1570f345a88b)
![IMG_0064](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/40360a5f-9ba8-41e2-a364-7b34e3ffcbaf)
![IMG_0065](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/5e827e40-f556-4a44-ba11-982e41f03bae)
![IMG_0066](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/4446aba2-c231-4de1-a7d2-51f31e3f4440)
![IMG_0067](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/46c66197-c1d7-498f-90af-d00a3ee1ed17)
![IMG_0068](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/04ef0f8c-9d68-47c7-996a-625c3a8da48d)
![IMG_0069](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/f794d998-6994-4677-a83b-64389568bec4)
![IMG_0070](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/4435aac6-c34e-4cb6-8f35-6e16212cb392)
![IMG_0071](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/70d75a7a-3048-4182-ad21-58018361d288)
![IMG_0072](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/a2946f95-4a31-4e8d-8013-bdb45408512a)

**Comment by AitorbnfFC** [comment]
That would be sublime🤞🏻🙏🏻

**Comment by neoalius** [comment]
Don't know if this would be helpful

https://developers.google.com/maps/documentation
https://developer.here.com/develop/rest-apis
[MC-10181818-0001.pdf](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11459391/MC-10181818-0001.pdf)

**Comment by djskual** [comment]
I don't have carnet activated.
Have a greyed checkbox in map view selection screen for google earth.
Hope this can be unlocked to use google earth in my car.

My car: VW Golf GTE 2016
My FW: MHI2_ER_VWG11_K3342_1_AIO_MU1427_20230318

**Comment by fhdoma** [comment]
I have MHI2_ER_SKG13_P4526 mib 2.5 FW 1440 enable Google earth will be perfect, in skoda cannot bui licence to Google earth :( ending in 2019 year

**Comment by neoalius** [comment]
I have a custom patch which enables it and point to paid server....Happy to share if that helps. The server would require and activation, so the same server can not be used...

**Reply by fhdoma** [reply]
Nice how can use with mib have some link or quide how use? Skoda ending possible buy licence in year 2019

**Reply by fhdoma** [reply]
@harman-f @neoalius can you to gather cooperation? Was by very useful, thank you

**Reply by Errivincenz** [reply]
> I have a custom patch which enables it and point to paid server....Happy to share if that helps. The server would require and activation, so the same server can not be used...

This could be interesting! I don't have time to start from nothing but, if you can share this patch, I'll try to study it and then dedicate myself to the realization of the server.

**Reply by fhdoma** [reply]
@Errivincenz @neoalius that's was by perfect

**Reply by fhdoma** [reply]
@Errivincenz hello do you have some files for implement this? If understand issue is server? I have choice but if click licence issue

**Comment by neoalius** [comment]
There were two packages delivered, 
Package 1
<img width="333" alt="Screenshot 2023-05-20 at 11 27 39 am" src="https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/bf609ef4-6973-412b-86c7-a936d4ec0882">

first one collected the details, Script as below 

=========================================

echo FinalScript for HW ${1} on medium ${2}...

cp ${2}/GoogleEarth/GEMIB/0/default/gemib /net/mmx/dev/shmem/gemib

on -f mmx /bin/chmod a+rwx /net/mmx/dev/shmem/gemib
on -f mmx /net/mmx/dev/shmem/gemib

echo Done.
touch /tmp/SWDLScript.Result
===============================================

Second One activated it. 
<img width="308" alt="Screenshot 2023-05-20 at 11 28 03 am" src="https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/131591423/3d76e4b7-0294-4a31-9425-446774d9e71e">

===============================================

#!/bin/ksh

echo FinalScript for HW ${1} on medium ${2}...

cp ${2}/GoogleEarth/GEMIB/0/default/gemib /net/mmx/dev/shmem/gemib

on -f mmx /bin/chmod a+rwx /net/mmx/dev/shmem/gemib
on -f mmx /net/mmx/dev/shmem/gemib

mount -u /net/rcc/mnt/efs-persist
rm -f /net/rcc/mnt/efs-persist/SWDL/FileCopyInfo/GoogleEarth.info

echo Done.
touch /tmp/SWDLScript.Result

=====================================================

**Reply by neoalius** [reply]
Costed me 150 EU

**Reply by harman-f** [reply]
not new and just the "small" part on unit. On the other end a full custom server is needed to handle the requests and send fitting data back to unit

**Reply by djskual** [reply]
ok, also, if you own a server, possible to create parameters to aswer to the mib and made earth working?
How?

**Reply by Mr-Chewie** [reply]
Can you upload the audimib and gemib files please?

**Reply by fhdoma** [reply]
@neoalius please

**Reply by Stick424** [reply]
@neoalius   Hey, can you send me the link where you buy it for 150€ plz ? Found someone but cost 350 ^^

**Reply by neoalius** [reply]
https://www.audi-mib.bg/audi-google-earth/

**Reply by fhdoma** [reply]
> https://www.audi-mib.bg/audi-google-earth/

Hello was by possible make USB update for mib 2.5 columbus škoda?

**Comment by djskual** [comment]
don't know if that can be usefull, but i've found this:
[https://github.com/jilleb/mib2-toolbox/issues/31](https://github.com/jilleb/mib2-toolbox/issues/31)

Seem usefull to activate 3D google earth map.

**Reply by harman-f** [reply]
Yes, that is still wprking (if you have connect running)

**Reply by fhdoma** [reply]
@harman-f but if I read this for work need licence Google earth but is no possible buy

**Reply by fhdoma** [reply]
@djskual but need licence and licence is not possible buy from 2019 in skoda

**Comment by proxynetul** [comment]
Any news ?

**Comment by LateAlways** [comment]
Sorry for bumping this inactive thread. I recently made this: https://github.com/LateAlways/mib-earth-bridge
It's open source too.

**Reply by Stick424** [reply]
Hello, thanks for the work man, how can I install it ? I put it on a USB Disk ?

**Reply by LateAlways** [reply]
It's kind of complicated. The main problem is the MMI checks for a license when booting the google earth view and you need to hook that with a patch like they make for navignore. Basically, the MMI asks audi connect whether you pay or not and satellitemaps will always return as not paid so you need to hook a ServiceListManager.java file and override the value to paid.

**Reply by fhdoma** [reply]
> Sorry for bumping this inactive thread. I recently made this: https://github.com/LateAlways/mib-earth-bridge It's open source too.

Hello is compatible with mib 2.5 skoda columbus?

Was by possible make USB updare? Thank you

**Reply by LateAlways** [reply]
This requires to host a server on your end on the web. You can't just install it with nothing else on the unit.

**Reply by 2p8h4ts44b-glitch** [reply]
Is it possible for you to provide step by step guide on how to do it? I have been trying to make it work for 2 weeks but no luck so if you can help it would be greatly appreciated.

**Reply by 2p8h4ts44b-glitch** [reply]
> It's kind of complicated. The main problem is the MMI checks for a license when booting the google earth view and you need to hook that with a patch like they make for navignore. Basically, the MMI asks audi connect whether you pay or not and satellitemaps will always return as not paid so you need to hook a ServiceListManager.java file and override the value to paid.

I was able to follow through step 2 and made changes but cant figure it what to do with step 3 which is ServiceListManager.java. would you mind explaiining how to hook this one up as I have full access to mmi but cant figure out this step.

**Reply by LateAlways** [reply]
I will provide more in detail instructions in the repo itself. I've got college and all so it might take a while.

**Reply by 2p8h4ts44b-glitch** [reply]
Thanks for reply. Sure no problem. whenever you can please let me know. Btw I am extermely close and I was able to grab servicelisthandler.class file from my car but cant figure out where to insert that code you have provided in step 3 in that file. whenever you have free time, let me know where to make changes it would be helpful. thanks again;)

**Reply by 2p8h4ts44b-glitch** [reply]
I think I was able to find out where to out the code and I uploaded the .jar file to car in jars folder but for some reason whenever i click on the google earth/satellite in maps it goes back to standard and i am not getting any incoming connection on my server either. I have Mhi2q_us_aug22_p5087 firmware version. so yeah, whenever you can, it would be helpful if you can provide in detail instructions on the repo in your free time. I will be waiting for it. thanks in advance.

**Reply by LateAlways** [reply]
Hey, so I'm actually working on a new version which wouldn't require running a server at all and would be as simple as running a script on your pc with some files from your unit and then upload the output to the unit using an automated script.

**Reply by LateAlways** [reply]
> I think I was able to find out where to out the code and I uploaded the .jar file to car in jars folder but for some reason whenever i click on the google earth/satellite in maps it goes back to standard and i am not getting any incoming connection on my server either. I have Mhi2q_us_aug22_p5087 firmware version. so yeah, whenever you can, it would be helpful if you can provide in detail instructions on the repo in your free time. I will be waiting for it. thanks in advance. 

Btw im pretty sure that you simply have to do delete all gecache in the GEM and do restart ge. It should be in production -> mmx_prod -> googleearth_something or gemib_something

**Reply by fhdoma** [reply]
Automatic script for mib 2.5 columbus on sd card without server was by perfect. Enabled Google earth in car and connect to server. Becouse this option have now disabled.

**Reply by Mr-MIBonk** [reply]
@LateAlways Feel free to contact me if you'd like; the serverless idea is excellent, and I'm sure we can implement it together 👍🏼

**Reply by fhdoma** [reply]
> @LateAlwaysNeváhejte kontaktovat@harman-fnebo mě, pokud chcete; nápad s bezserverovým systémem je vynikající a jsem si jistý, že ho dokážeme společně implementovat 👍🏼

Hello

If found good solution please send me.

Thank you

**Reply by LateAlways** [reply]
Its actually really simple I believe. You would only have to change the device type used in the libearthmobile.so library to something other than QNX and google should stop blocking requests. Then, idk if the car knows how to request dns for kh.google.com but that's another story.

**Reply by Mr-MIBonk** [reply]
I know, @harman-f did it today with lokal patch, same idea... but what is with "ServiceListManager.java"?
I think we need some general patch in the Servicelist...

**Reply by harman-f** [reply]
Not tested yet, but this looks promising:

File: /mnt/app/gemib/lib/libearthmobile.so

Patch purpose: restore direct Google Earth communication without an external proxy.

Patch positions:

* 0x0133C8DC: 51 4E 58 → 71 6E 78 (QNX → qnx) — avoids Google’s case-sensitive QNX User-Agent block.
* 0x004C16A4: 05 10 84 E0 → 3C FF FF EA — suppresses the obsolete type=embedded dbRoot parameter.

**Reply by harman-f** [reply]
@LateAlways 
Can you point a direction to ServiceListManager.java

**Reply by LateAlways** [reply]
I was wrong all along. It is inside de.audi.tghu.navi.app.online.ServiceList**Handler**

**Reply by harman-f** [reply]
we are on Skoda... having trouble to find a similar function so far...

**Reply by harman-f** [reply]
@LateAlways 
Which exact JAR/bundle contains de.audi.tghu.navi.app.online.ServiceListHandler on your unit, and which MHI2 train/brand was that dump from?

**Reply by LateAlways** [reply]
MHS2_US_AU_P1242, inside the lsd.jxe.

**Reply by harman-f** [reply]
That is quite change in direction from looking for a MHI2 solution

**Reply by LateAlways** [reply]
It shouldnt be that hard. There are plenty of other places where a hook like this can be implemented.

**Reply by harman-f** [reply]
I'm trying now.

You should add a train info to your repo.

**Reply by MattGerrand** [reply]
> Not tested yet, but this looks promising:
> 
> File: /mnt/app/gemib/lib/libearthmobile.so
> 
> Patch purpose: restore direct Google Earth communication without an external proxy.
> 
> Patch positions:
> 
> * 0x0133C8DC: 51 4E 58 → 71 6E 78 (QNX → qnx) — avoids Google’s case-sensitive QNX User-Agent block.
> * 0x004C16A4: 05 10 84 E0 → 3C FF FF EA — suppresses the obsolete type=embedded dbRoot parameter.

@harman-f, did you ever test this? Thinking about trying this patch on MHI2Q_ER_AUG22_P3638 - MU0917.

**Comment by chefranov** [comment]
I use HERE and own server 
<img width="1291" height="589" alt="image" src="https://github.com/user-attachments/assets/bca5055f-4ff9-46fa-8e0e-68d7baeb66c9" />
